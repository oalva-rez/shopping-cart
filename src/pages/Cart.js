import React from "react";

export default function Cart(props) {
  return (
    <div className="cart">
      <h1>Cart</h1>
      <div className="cart-items">
        {props.cart.map((item) => {
          return (
            <div className="item" key={item.item.id}>
              <img src="" alt="" />
              <div className="item-details">
                <h3>{item.item.title}</h3>
                <p>Description: {item.item.description}</p>
                <p>Size: {item.item.size ? item.item.size : "One Size"}</p>
                <p>Price: {item.item.price}</p>
                <p>x{item.quantity}</p>
              </div>
              <div className="item-price">
                {parseFloat(item.item.price * item.quantity).toFixed(2)}
              </div>
              <button
                onClick={() => {
                  props.deleteItem(item.item.id);
                }}
              >
                X
              </button>
            </div>
          );
        })}
      </div>
    </div>
  );
}
