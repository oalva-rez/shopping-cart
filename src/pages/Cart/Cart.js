import React from "react";
import { Link } from "react-router-dom";

export default function Cart(props) {
  return props.cart.length > 0 ? (
    <div className="cart">
      <h1 className="cart-heading">Cart</h1>
      <div className="cart-items">
        {props.cart.map((item) => {
          return (
            <div className="item" key={item.item.id}>
              <img src={item.item.image} alt="" />
              <div className="item-details">
                <h3>{item.item.title}</h3>
                <p>
                  <span className="bold">Description:</span>{" "}
                  {item.item.description}
                </p>
                <p>
                  <span className="bold">Size:</span>{" "}
                  {item.item.size ? item.item.size : "One Size"}
                </p>
                <p>
                  <span className="bold">Price:</span> {item.item.price} x
                  {item.quantity}
                </p>
              </div>
              <div className="item-price">
                ${parseFloat(item.item.price * item.quantity).toFixed(2)}
              </div>
              <button
                className="remove-button"
                onClick={() => {
                  props.deleteItem(item.item.id);
                }}
              >
                X
              </button>
            </div>
          );
        })}
      </div>
      <div className="cart-total"></div>
    </div>
  ) : (
    <div className="cart-empty">
      <h1 className="cart-heading">Cart</h1>
      <h2 className="empty-msg">
        Your cart is currently empty. <Link to="/products">keep shopping.</Link>
      </h2>
    </div>
  );
}
