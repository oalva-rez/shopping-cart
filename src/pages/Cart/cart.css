.cart-heading {
  text-align: center;
}
.empty-msg {
  text-align: center;
  display: grid;
  place-content: center;
  height: 50vh;
}
.empty-msg a {
  color: black;
  font-style: italic;
}
.cart-items {
  width: 70%;
  margin: 30px auto;
}
.cart-items img {
  height: 142px;
  width: 100px;
}
.cart-items .item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 20px;
}
.cart-items .item:nth-child(even) {
  background-color: rgb(246, 246, 246);
}

.cart-items .item-details {
  width: 70%;
}
.cart-items .item-details p {
  margin-bottom: 10px;
}
.bold {
  font-weight: 600;
}
.remove-button {
  background: #ff7d73;
  color: #fff;
  border: none;
  padding: 5px 10px;
  border-radius: 50%;
  cursor: pointer;
}
.cart {
  min-height: 80vh;
}
