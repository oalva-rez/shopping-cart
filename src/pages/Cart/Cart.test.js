import { render, screen } from "@testing-library/react";
import Cart from "./Cart";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";

const mockDeleteItem = jest.fn();
const MockEmptyCart = () => (
  <BrowserRouter>
    <Cart cart={[]} deleteItem={mockDeleteItem} />
  </BrowserRouter>
);
const mockCart = [
  {
    item: {
      id: 1,
      title: "Product 1",
      description: "This is a product",
      price: "10.00",
      image: "https://picsum.photos/200",

      size: "S",
    },
    quantity: 1,
  },
  {
    item: {
      id: 2,
      title: "Product 2",
      description: "This is a product",
      price: "10.00",
      image: "https://picsum.photos/200",
      size: "S",
    },
    quantity: 2,
  },
];

describe("Cart with items", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <Cart deleteItem={mockDeleteItem} cart={mockCart} />
    );
    expect(container).toMatchSnapshot();
  });
  it("renders cart items", () => {
    render(<Cart deleteItem={mockDeleteItem} cart={mockCart} />);
    expect(screen.getByText("Product 1")).toBeInTheDocument();
    expect(screen.getByText("Product 2")).toBeInTheDocument();
  });
  it('renders heading "Cart"', () => {
    render(<Cart deleteItem={mockDeleteItem} cart={mockCart} />);
    expect(screen.getByRole("heading", { name: "Cart" })).toBeInTheDocument();
  });
  it("renders cart total", () => {
    render(<Cart deleteItem={mockDeleteItem} cart={mockCart} />);
    expect(screen.getByText("Total: $30.00")).toBeInTheDocument();
  });
  it("renders correct item total", () => {
    render(<Cart deleteItem={mockDeleteItem} cart={mockCart} />);
    expect(screen.getByText("$20.00")).toBeInTheDocument();
  });
  it("deletes item from cart", () => {
    render(<Cart deleteItem={mockDeleteItem} cart={mockCart} />);

    const deleteButtons = screen.getAllByRole("button", { name: "X" });
    deleteButtons.forEach((button) => {
      userEvent.click(button);
    });
    expect(mockDeleteItem).toHaveBeenCalledTimes(mockCart.length);
  });
});

describe("Empty items", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <MockEmptyCart deleteItem={mockDeleteItem} cart={[]} />
    );
    expect(container).toMatchSnapshot();
  });
  it('renders heading "Cart"', () => {
    render(<MockEmptyCart deleteItem={mockDeleteItem} cart={[]} />);
    expect(screen.getByRole("heading", { name: "Cart" })).toBeInTheDocument();
  });
  it("renders cart items", () => {
    render(<MockEmptyCart deleteItem={mockDeleteItem} cart={[]} />);
    expect(
      screen.getByText("Your cart is currently empty.")
    ).toBeInTheDocument();
  });
  it("renders link to products page", () => {
    render(<MockEmptyCart deleteItem={mockDeleteItem} cart={[]} />);
    expect(screen.getByText("keep shopping.")).toBeInTheDocument();
    expect(screen.getByTestId("keep-shopping-link")).toHaveAttribute(
      "href",
      "/products"
    );
  });
});
