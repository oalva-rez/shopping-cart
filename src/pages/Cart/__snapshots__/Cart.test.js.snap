// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart with items matches snapshot 1`] = `
<div>
  <div
    class="cart"
  >
    <h1
      class="cart-heading"
    >
      Cart
    </h1>
    <div
      class="cart-items"
    >
      <div
        class="item"
      >
        <img
          alt=""
          src="https://picsum.photos/200"
        />
        <div
          class="item-details"
        >
          <h3>
            Product 1
          </h3>
          <p
            class="cart-desc"
          >
            <span
              class="bold"
            >
              Description:
            </span>
             
            This is a product
          </p>
          <p>
            <span
              class="bold"
            >
              Size:
            </span>
             
            S
          </p>
          <p>
            <span
              class="bold"
            >
              Price:
            </span>
             $
            10.00
          </p>
          <p>
            <span
              class="bold"
            >
              Qty: 
            </span>
            1
          </p>
        </div>
        <div
          class="item-price"
        >
          $
          10.00
        </div>
        <button
          class="remove-button"
        >
          X
        </button>
      </div>
      <div
        class="item"
      >
        <img
          alt=""
          src="https://picsum.photos/200"
        />
        <div
          class="item-details"
        >
          <h3>
            Product 2
          </h3>
          <p
            class="cart-desc"
          >
            <span
              class="bold"
            >
              Description:
            </span>
             
            This is a product
          </p>
          <p>
            <span
              class="bold"
            >
              Size:
            </span>
             
            S
          </p>
          <p>
            <span
              class="bold"
            >
              Price:
            </span>
             $
            10.00
          </p>
          <p>
            <span
              class="bold"
            >
              Qty: 
            </span>
            2
          </p>
        </div>
        <div
          class="item-price"
        >
          $
          20.00
        </div>
        <button
          class="remove-button"
        >
          X
        </button>
      </div>
    </div>
    <div
      class="cart-total"
    >
      <p>
        Total: $
        30.00
      </p>
      <button>
        Checkout
      </button>
    </div>
  </div>
</div>
`;

exports[`Empty cart matches snapshot 1`] = `
<div>
  <div
    class="cart-empty"
  >
    <h1
      class="cart-heading"
    >
      Cart
    </h1>
    <h2
      class="empty-msg"
    >
      Your cart is currently empty.
       
      <a
        data-testid="keep-shopping-link"
        href="/products"
      >
        keep shopping.
      </a>
    </h2>
  </div>
</div>
`;
